docs/content/specs/terraform/_index.md

```diff
@@ -533,6 +543,466 @@ Bad examples:
+EOT
+}
+```
+
+<br>
+
+---
+
+<br>
+
+#### ID: TFNFR18 - Category: Code Style - Every `variable` **MUST** have an appropriate `type`
+
+`type` **MUST** be defined for every `variable`. `type` should be as precise as possible, `any` can only be defined with adequate reasons.
+
+* Use `bool` instead of `string` or `number` for `true/false`
+* Use `string` for text
+* Use concrete `object` instead of `map(any)`
+
+<br>
+
+---
+
+<br>
+
+#### ID: TFNFR19 - Category: Code Style - `variable` containing confidential data should be declared as `sensitive = true`
+
+If `variable`'s `type` is `object` and contains one or more fields that would be assigned to a `sensitive` argument, then this whole `variable` should be declared as `sensitive = true`, otherwise you should extract sensitive field into separated variable block with `senstive = true`.
+
+<br>
+
+---
+
+<br>
+
+#### ID: TFNFR20 - Category: Code Style - Declare `nullable = false` when it's possible
+
+<br>
+
+---
+
+<br>
+
+#### ID: TFNFR21 - Category: Code Style - **MUST NOT** declare `nullable = true`
+
+<br>
+
+---
+
+<br>
+
+#### ID: TFNFR22 - Category: Code Style - **MUST NOT** declare `sensitive = false`
+
+<br>
+
+---
+
+<br>
+
+#### ID: TFNFR23 - Category: Code Style - `variable` with `sensitive = true` **MUST NOT** have default value unless the default value represents turning off a feature, like `default = null` or `default = []`
+
+<br>
+
+---
+
+<br>
+
+#### ID: TFNFR24 - Category: Code Style - Deal with deprecated `variable`
+
+Sometimes we will find names for some `variable` are not suitable anymore, or a change should be made to the data type. We want to ensure forward compatibility within a major version, so direct changes are strictly forbidden. The right way to do this is move this `variable` to an independent `deprecated_variables.tf` file, then redefine the new parameter in `variable.tf` and make sure it's compatible everywhere else.
+
+Deprecated `variable` must be annotated as `DEPRECATED` at the beginning of the `description`, at the same time the replacement's name should be declared. E.g.
+
+```terraform
+variable "enable_network_security_group" {
+  type        = string
+  default     = null
+  description = "DEPRECATED, use `network_security_group_enabled` instead; Whether to generate a network security group and assign it to the subnet. Changing this forces a new resource to be created."
+}
+```
+
+A cleanup of `deprecated_variables.tf` can be performed during a major version release.
+
+<br>
+
+---
+
+<br>
+
+#### ID: TFNFR25 - Category: Code Style - All verified modules **MUST** have `version.tf` file
+
+`versions.tf` file can only contain one `terraform` block.
```

Comment:
Currently we call this file `terraform.tf`.

Would you be prepared to change the name to match the existing standard?